### PR TITLE
building design matrix and response arrays

### PR DESCRIPTION
**Pull request by jkim0731**

### CODE REVIEW

File: code/capsule_utils.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+def add_trials_to_bod(bod, response_window=[0.15, 0.75]):
+    """ Temporary fix to add trials to bod
+    using stimulus_presentations and licks.
+    No correct rejection for this.
+    Columns: 'change_time', 'hit', 'miss'
+
+    Parameters
+    ----------
+    bod : BehaviorOphysDataset
```
Comment (yavorska-iryna) [inline code/capsule_utils.py:24]: Is this something that exists in COMB and has the same structure as SDK object?

File: code/capsule_utils.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    for session_name,session_dict in capsule_files.items():
+        # load session_json from session_path
+        with open(session_dict['raw_path'] / "session.json", 'r') as f:
+            session_json = json.load(f)
+        dict_map[session_name]  = session_json['session_type']
+        
+    return dict_map
+
+################################################################################################
+# Functions for data access - FROM OTHER REPOS SHOULD DELETE
```
Comment (yavorska-iryna) [inline code/capsule_utils.py:64]: Are any of the functions below going to be rewritten or reused?

File: code/capsule_utils.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+    if sync_file_path is None:
+        # method 2: load platform json
+        # Note: sometimes fails if platform json has incorrect sync_file path
+        logging.info(f"Trying to find sync file using platform json for {input_path}")
+        file_parts = {"platform_json": "_platform.json"}
+        platform_path = find_data_file(input_path, file_parts["platform_json"])
+        with open(platform_path, 'r') as f:
+            platform_json = json.load(f)
+        ophys_folder = check_ophys_folder(input_path)
```
Comment (yavorska-iryna) [inline code/capsule_utils.py:173]: is sync file saved in ophys folder? does it sync behavior videos? (sorry I should know that)

File: code/kernel_tools.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    run_params['kernels'] = kernels
+    run_params['dropouts'] = dropouts
+    return run_params
+
+
+def replace_kernels(kernels, bod):
+    '''
+        Replaces the 'each-image' kernel with each individual image (not omissions), with the same parameters
+    '''
+    if ('each-image' in kernels) & ('any-image' in kernels):
```
Comment (yavorska-iryna) [inline code/kernel_tools.py:51]: maybe in the future we can change "image" for "stimulus" for generalizability

**Review by yavorska-iryna — APPROVED** [review]

File: code/capsule_utils.py (hunk 1 above)
Comment (jkim0731) [reply code/capsule_utils.py:24]: Yes, exactly!

File: code/capsule_utils.py (hunk 2 above)
Comment (jkim0731) [reply code/capsule_utils.py:64]: Yes I think so. Currently I'm only following @mattjdavis's capsule - https://codeocean.allenneuraldynamics.org/capsule/1645683/tree. My guess is that when COMB is fully updated, these functions will be obsolete (or some of them are already obsolete).

File: code/capsule_utils.py (hunk 3 above)
Comment (jkim0731) [reply code/capsule_utils.py:173]: Yes I think so

**Pull request merged**